test/hexpm_web/controllers/package_report_controller_test.exs

```diff
@@ -0,0 +1,305 @@
+defmodule HexpmWeb.PackageReportControllerTest do
+  use HexpmWeb.ConnCase, async: true
+
+  setup do
+    user1 = insert(:user) # Package 1 owner
+    user2 = insert(:user, role: "moderator") # Pacakge 2 owner and mod
+    user3 = insert(:user) # Report 1 author
+    user4 = insert(:user) # No author, no mod, no owner
+
+    repository1 = insert(:repository)
+    repository2 = insert(:repository)
+
+    owners = [build(:package_owner, user: user1)]
+    package1 = insert(:package, name: "package1", package_owners: owners)
+
+    package2 = insert(:package, owners: [user2])
+
+    report1 = insert(
+      :package_report,
+      package: package1,
+      author: user3,
+      state: "to_accept",
+      description: "report for first package"
+    )
+
+    report2 = insert(
+      :package_report,
+      package: package1,
+      author: user3,
+      state: "accepted",
+      description: "report for first package"
+    )
+
+    report3 = insert(
+      :package_report,
+      package: package1,
+      author: user3,
+      state: "solved",
+      description: "report for first package"
+    )
+
+    report4 = insert(
+      :package_report,
+      package: package1,
+      author: user3,
+      state: "rejected",
+      description: "report for first package"
+    )
+
+    insert(
+      :release,
+      package: package1,
+      version: "0.0.1",
+      meta: build(:release_metadata, app: package1.name)
+    )
+
+    insert(
+      :release,
+      package: package1,
+      version: "0.0.2",
+      meta: build(:release_metadata, app: package1.name)
+    )
+
+    insert(
+      :release,
+      package: package1,
+      version: "0.0.3-dev",
+      meta: build(:release_metadata, app: package1.name)
+    )
+
+    insert(
+      :release,
+      package: package2,
+      version: "1.0.0",
+      meta: build(:release_metadata, app: package2.name)
+    )
+
+    insert(:organization_user, user: user1, organization: repository1.organization)
+
+    %{
+      package1: package1,
+      package2: package2,
+      report1: report1,
+      report2: report2,
+      report3: report3,
+      report4: report4,
+      repository1: repository1,
+      repository2: repository2,
+      user1: user1,
+      user2: user2,
+      user3: user3,
+      user4: user4
+    }
+  end
+
+  describe "GET /reports" do
+    test "list all", %{user1: user1, report1: report1} do
+      conn =
+        build_conn()
+        |> test_login(user1)
+        |> get("/reports")
+
+      result = response(conn, 200)
+      assert result =~ ~r/#{report1.id}/s
```

Comment:
Wrap this in the HTML tags around the id. For example if the id is `1` it is likely it will match even if there is failure since we likely have the number `1` somewhere in all rendered pages.

Reply:
You don't have to use a regex here `assert result =~ "#{report1.id}"` is enough.